frame/support/test/tests/pallet_ui/storage_ensure_span_are_ok_on_wrong_gen.stderr

```diff
@@ -0,0 +1,33 @@
+error[E0277]: the trait bound `Bar: WrapperTypeDecode` is not satisfied
+  --> $DIR/storage_ensure_span_are_ok_on_wrong_gen.rs:20:12
+   |
+20 |     #[pallet::storage]
```

Comment:
Not such a good diagnostic. Can we not use the correct span here?

Reply:
yes we can point to the type alias instead of the storage attribute.

note that this not due to this PR, it is because we use the span of the attribute when expanding the storage metadata and getter. I think we can do in a follow-up i'll do an issue and look at it.

Reply:
I tried to improve a bit in this draft https://github.com/paritytech/substrate/pull/8850

Reply:
Perfect! :)